packages/device-connector/src/device-types/store.ts

```diff
@@ -6,12 +6,24 @@ import { logger } from '../util/logger'
 export const DeviceTypes = {
   store: new Array<DeviceType>(),
 
-  withFQBN (fqbn: FQBN): DeviceType | undefined {
-    return this.store.find((item) => item.fqbn === fqbn)
+  withFQBN (fqbn: FQBN): DeviceType {
+    const devType = this.store.find((item) => item.fqbn === fqbn)
+
+    if (devType == null) {
+      throw new Error(`No DeviceType with fqbn ${fqbn} found locally`)
+    }
+
+    return devType
   },
 
-  withId (typeId: string): DeviceType | undefined {
-    return this.store.find((item) => item.id === typeId)
+  withId (typeId: string): DeviceType {
```

Comment:
(find)byId, see above

Reply:
refactored in 37f119b16a9a2ece04526ab67f65e60611370b7b